atomicfu-gradle-plugin/src/main/kotlin/kotlinx/atomicfu/plugin/gradle/AtomicFUGradlePlugin.kt

```diff
@@ -337,8 +337,9 @@ private fun Project.configureTransformationForTarget(target: KotlinTarget) {
                 it.destinationDirectory.value(originalDestinationDirectory)
             }
         }
-        val originalClassesDirs: FileCollection =
-            project.files(classesDirs.from.toTypedArray()).filter { it.exists() }
+        val originalClassesDirs: FileCollection = 
+            project.objects.fileCollection().from(compilationTask.flatMap { it.destinationDirectory }) + 
+                project.files(classesDirs.from.toTypedArray()).filter { it.exists() }
```

Comment:
Nit: You could call multiple times `.from` to add more things into `ConfigurableFileCollection`:

```suggestion
        val originalClassesDirs: FileCollection = project.objects.fileCollection()
            .from(compilationTask.flatMap { it.destinationDirectory })
            .from({ project.files(classesDirs.from).filter { it.exists() } })
```